src/tokens/TokensService.ts

```diff
@@ -74,19 +81,24 @@ export class TokensService {
   async getLiquidtyPoolTokenBalances(sett: SettDefinition, settSnapshot: SettSnapshot): Promise<TokenBalance[]> {
     const { depositToken, protocol } = sett;
 
-    let poolData;
+    let poolData: SushiswapPairQuery | UniswapPairQuery | undefined;
     if (protocol === 'uniswap') {
```

Comment:
you think we should keep this as an if list? we an also use `Protocol` enum here that were introduced in `util` of all places naturally

Reply:
For now, I think leaving this block as an if statement for this diff is alright since the compiler evaluates the `Protocol` enum members to make sure this check is type-safe. That being said, I think doing this casing behavior can get a little hard to maintain so we can construct a Protocol entity in the future which abstracts away the type and implementation on how to fetch the pairs. 

What do you think?

Reply:
yeah, that sounds good.